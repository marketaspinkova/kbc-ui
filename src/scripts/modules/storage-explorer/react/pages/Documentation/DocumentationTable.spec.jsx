import React from 'react';
import DocumentationTable from './DocumentationTable';
import { fromJS, Map } from 'immutable';

const documentationTree = fromJS([
  {
    id: 'bucket1',
    bucketDescription: 'this is bucket1 description',
    bucketTables: [
      {
        name: 'table1',
        id: 'bucket1.table1',
        tableDescription: 'this is table1 description',
        columnsDescriptions: {
          column1: 'this is column1 description',
          column2: null,
          column3: 'column3 description blabla ###heading blabla'
        }
      },
      {
        name: 'table2',
        id: 'bucket1.table2',
        tableDescription: null,
        columnsDescriptions: {
          column1: null,
          column2: null,
          column3: null
        }
      }
    ]
  },
  {
    id: 'emptyBucketNoDescription',
    bucketDescription: null,
    bucketTables: []
  },
  {
    id: 'emptyBucket',
    bucketDescription: 'this is emptyBucket description',
    bucketTables: []
  }
]);

const emptyDocumentationTree = fromJS([]);

describe('<DocumentationTable />', function() {
  it('should render empty documentation tree', function() {
    shallowSnapshot(
      <DocumentationTable
        documentationTree={emptyDocumentationTree}
        isSearchQuery={false}
        openedRows={Map()}
        toggleDocumentationRow={() => null}
      />
    );
  });

  it('should render buckets only documentation tree', function() {
    shallowSnapshot(
      <DocumentationTable
        documentationTree={documentationTree}
        isSearchQuery={false}
        openedRows={Map()}
        toggleDocumentationRow={() => null}
      />
    );
  });

  it('should render whole documentation tree', function() {
    shallowSnapshot(
      <DocumentationTable
        documentationTree={documentationTree}
        isSearchQuery={true}
        openedRows={Map()}
        toggleDocumentationRow={() => null}
      />
    );
  });

  it('should render toggled bucket documentation tree', function() {
    shallowSnapshot(
      <DocumentationTable
        documentationTree={documentationTree}
        isSearchQuery={false}
        openedRows={fromJS({
          BUCKET_ROWbucket1: true
        })}
        toggleDocumentationRow={() => null}
      />
    );
  });

  it('should render toggled bucket and table documentation tree', function() {
    shallowSnapshot(
      <DocumentationTable
        documentationTree={documentationTree}
        isSearchQuery={false}
        openedRows={fromJS({
          BUCKET_ROWbucket1: true,
          'TABLE_ROWbucket1.table1': true
        })}
        toggleDocumentationRow={() => null}
      />
    );
  });

  it('should render whole tree if has toggled bucket and table', function() {
    shallowSnapshot(
      <DocumentationTable
        documentationTree={documentationTree}
        isSearchQuery={true}
        openedRows={fromJS({
          BUCKET_ROWbucket1: true,
          TABLE_ROWtable1: true
        })}
        toggleDocumentationRow={() => null}
      />
    );
  });
});
