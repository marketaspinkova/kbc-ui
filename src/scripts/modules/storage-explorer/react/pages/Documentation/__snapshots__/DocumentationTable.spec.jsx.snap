// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentationTable /> should render buckets only documentation tree 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={true}
>
  <tbody>
    <tr
      key="bucket1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          bucket1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is bucket1 description"
        />
      </td>
    </tr>
    <tr
      key="emptyBucketNoDescription"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucketNoDescription
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucket"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucket
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is emptyBucket description"
        />
      </td>
    </tr>
  </tbody>
</Table>
`;

exports[`<DocumentationTable /> should render empty documentation tree 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={true}
>
  <tbody />
</Table>
`;

exports[`<DocumentationTable /> should render toggled bucket and table documentation tree 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={true}
>
  <tbody>
    <tr
      key="bucket1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-down"
          />
           
          <i
            className="fa fa-folder"
          />
           
          bucket1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is bucket1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer table-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-down"
          />
           
          <i
            className="fa fa-table"
          />
           
          table1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is table1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1column1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is column1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1column2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table1column3"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column3
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="column3 description blabla ###heading blabla"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer table-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-table"
          />
           
          table2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucketNoDescription"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucketNoDescription
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucket"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucket
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is emptyBucket description"
        />
      </td>
    </tr>
  </tbody>
</Table>
`;

exports[`<DocumentationTable /> should render toggled bucket documentation tree 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={true}
>
  <tbody>
    <tr
      key="bucket1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-down"
          />
           
          <i
            className="fa fa-folder"
          />
           
          bucket1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is bucket1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer table-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-table"
          />
           
          table1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is table1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer table-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-table"
          />
           
          table2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucketNoDescription"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucketNoDescription
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucket"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="kbc-cursor-pointer bucket-row"
          onClick={[Function]}
        >
          <i
            className="fa fa-caret-right"
          />
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucket
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is emptyBucket description"
        />
      </td>
    </tr>
  </tbody>
</Table>
`;

exports[`<DocumentationTable /> should render whole documentation tree 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={true}
>
  <tbody>
    <tr
      key="bucket1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="bucket-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-folder"
          />
           
          bucket1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is bucket1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="table-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-table"
          />
           
          table1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is table1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1column1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is column1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1column2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table1column3"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column3
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="column3 description blabla ###heading blabla"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="table-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-table"
          />
           
          table2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table2column1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table2column2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table2column3"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column3
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucketNoDescription"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="bucket-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucketNoDescription
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucket"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="bucket-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucket
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is emptyBucket description"
        />
      </td>
    </tr>
  </tbody>
</Table>
`;

exports[`<DocumentationTable /> should render whole tree if has toggled bucket and table 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={true}
>
  <tbody>
    <tr
      key="bucket1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="bucket-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-folder"
          />
           
          bucket1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is bucket1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="table-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-table"
          />
           
          table1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is table1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1column1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is column1 description"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table1column2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table1column3"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column3
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="column3 description blabla ###heading blabla"
        />
      </td>
    </tr>
    <tr
      key="bucket1.table2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="table-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-table"
          />
           
          table2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table2column1"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column1
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table2column2"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column2
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="bucket1.table2column3"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="column-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-columns"
          />
           
          column3
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucketNoDescription"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="bucket-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucketNoDescription
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        N/A
      </td>
    </tr>
    <tr
      key="emptyBucket"
    >
      <td
        className="text-nowrap"
      >
        <div
          className="bucket-row"
          onClick={[Function]}
        >
           
          <i
            className="fa fa-folder"
          />
           
          emptyBucket
        </div>
      </td>
      <td
        className="kbc-break-all kbc-break-word"
      >
        <Markdown
          collapsible={true}
          size="normal"
          source="this is emptyBucket description"
        />
      </td>
    </tr>
  </tbody>
</Table>
`;
